DiscountDiscover/AppDelegate.m

```diff
@@ -87,6 +87,21 @@ -(void)scheduleNotification:(Deal *)deal {
     [center addNotificationRequest:request withCompletionHandler:nil];
 }
 
+-(void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
+    
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    DetailsViewController *detailsViewController = [storyboard instantiateViewControllerWithIdentifier:@"DetailsViewController"];
+    
+    NSData *dealData = response.notification.request.content.userInfo[@"Deal"];
+    Deal *deal = [NSKeyedUnarchiver unarchivedObjectOfClass:Deal.class fromData:dealData error:nil];
+    detailsViewController.deal = deal;
```

Comment:
You should check if `deal != nil` before instantiating and pushing the view controller to be safe.